### PR TITLE
Revert: REF: do length-checks in boilerplate decorator

### LINKED ISSUE #34123: BUG: Index with tuples comparison not working anymore
It seems trace back to #34081

with Index we could do something like:
```python
>>> pd.Index(['a', 'b', 'c']) == 'c'
array([False, False,  True])
```
And it's also possible to do the same for Index with tuples:
```python
>>> pd.Index([('a', 'b'), ('b', 'c'), ('c', 'a')]) == ('c', 'a')
array([False, False, True])
```
But seems not working this way anymore after #34081 , is this length check here correct (so it is designed this way on purpose)? or we should consider it a bug?

cc @jbrockmendel 

currently master:
```python
>>> pd.Index([('a', 'b'), ('b', 'c'), ('c', 'a')]) == ('c', 'a')
ValueError: ('Lengths must match', (3,), (2,), <class 'tuple'>)
```

### PR DESCRIPTION
**Pull request by charlesdong1991**
closes #34123 

As discussed in #34123 
let's revert this PR34081 first since it has potentially big impact on some cases in ops.

cc @jbrockmendel @jorisvandenbossche

### CODE REVIEW

File: pandas/core/arrays/timedeltas.py [hunk 1]
```diff
@@ -460,6 +460,10 @@ def __mul__(self, other):
         if not hasattr(other, "dtype"):
             # list, tuple
             other = np.array(other)
+        if len(other) != len(self) and not is_timedelta64_dtype(other):
```
Comment (jbrockmendel) [inline pandas/core/arrays/timedeltas.py:463]: other -> other.dtype

**Comment by jbrockmendel** [comment]
small request, otherwise LGTM

File: pandas/core/arrays/timedeltas.py (hunk 1 above)
Comment (charlesdong1991) [reply pandas/core/arrays/timedeltas.py]: done, thanks!

**Comment by jbrockmendel** [comment]
Can you add test(s) for the behavior that was broken in #34081

**Comment by charlesdong1991** [comment]
emm, interesting, somehow now on master, 
```python
>>> pd.Index([('a', 'b'), ('b', 'c'), ('c', 'a')]) == ('c', 'a')
array([False, False, False])
```

@jbrockmendel any idea?

**Comment by charlesdong1991** [comment]
emm, something might have changed alongside, now after this revert pr, 
```python
>>> pd.Index([('a', 'b'), ('b', 'c'), ('c', 'a')]) == ('c', 'a')
array([False, False, False])
```
it returns all False.

@jbrockmendel any idea which PRs recently merged could influence this?

**Comment by charlesdong1991** [comment]
emm, not sure how on earth i could run get `array([False, False, True])` result that day, cannot remember which version i used. but seems since 1.0 or later this case returns `array([False, False, False])` which i feel is wrong. i put some comments and todo there and pls let me know if it is appropriate

**Comment by jorisvandenbossche** [comment]
@charlesdong1991 I was also quite sure that I saw the correct behaviour in one of my environments, but also don't know where anymore.

Now, looking at it on 1.0.3, the problem is that it is creating a MultiIndex, and not a plain index with tuples in it. Ensuring we get an Index, I actually get the correct behaviour:

```
In [14]: pd.Index([('a', 'b'), ('b', 'c'), ('c', 'a')], tupleize_cols=False) == ('c', 'a') 
Out[14]: array([False, False,  True])
```

(and this example also actually fails on master)

**Comment by charlesdong1991** [comment]
yeah, indeed, the issue i saw is that in such case, it automatically recognises as a MI, not Index. 

@jorisvandenbossche thanks very much, I will add `tupleize_cols=False` in the test to get the correct result and I will also create an issue to report this case.

**Comment by jbrockmendel** [comment]
LGTM

**Comment by jreback** [comment]
thanks @charlesdong1991 

can you open a new issue to re-implement this (e.g. essentially a cleanup to revert the revert with appropriate changes).

**Comment by charlesdong1991** [comment]
ref #34233 

cc. @jreback

**Pull request merged**